app/pb_kits/playbook/pb_layout/layout.rb

```diff
@@ -2,92 +2,47 @@
 
 module Playbook
   module PbLayout
-    class Layout < Playbook::PbKit::Base
-      PROPS = %i[configured_classname
-                 configured_collapse
-                 configured_dark
-                 configured_data
-                 configured_full
-                 configured_id
-                 configured_position
-                 configured_transparent
-                 configured_size
-                 block].freeze
+    class Layout
+      include Playbook::Props
 
-      def initialize(classname: default_configuration,
-                     collapse: default_configuration,
-                     dark: default_configuration,
-                     data: default_configuration,
-                     full: default_configuration,
-                     id: default_configuration,
-                     position: default_configuration,
-                     transparent: default_configuration,
-                     size: default_configuration,
-                     &block)
-        self.configured_classname = classname
-        self.configured_collapse = collapse
-        self.configured_dark = dark
-        self.configured_data = data
-        self.configured_full = full
-        self.configured_id = id
-        self.configured_position = position
-        self.configured_transparent = transparent
-        self.configured_size = size
-        self.block = block_given? ? block : nil
-      end
-
-      def position
-        if configured_position == default_configuration
-          "_left"
-        else
-          "_#{configured_position}"
-        end
-      end
+      partial "pb_layout/layout"
 
-      def transparent
-        true_value(configured_transparent, "_transparent", "")
+      prop :collapse, type: Playbook::Props::Enum,
+                      values: %w[xs sm md lg xl],
+                      default: "xs"
+      prop :dark, type: Playbook::Props::Boolean, default: false
+      prop :full, type: Playbook::Props::Boolean, default: false
+      prop :position, type: Playbook::Props::Enum,
+                      values: %w[left right],
+                      default: "left"
+      prop :transparent, type: Playbook::Props::Boolean, default: false
+      prop :size, type: Playbook::Props::Enum,
+                      values: %w[xs sm md lg xl],
+                      default: "md"
+      def dark_class
+        dark ? "dark" : nil
       end
 
-      def size
-        if configured_size == default_configuration
-          "md"
-        else
-          configured_size
-        end
+      def full_class
+        full ? " full" : ""
       end
 
-      def collapse
-        if configured_collapse == default_configuration
-          " layout#{position}_collapse_xs"
-        else
-          " layout#{position}_collapse_#{configured_collapse}"
-        end
+      def transparent_class
+        transparent ? "transparent" : nil
       end
 
-      def full
-        true_value(configured_full, " full", "")
+      def collapse_class
```

Comment:
Instead of putting whitespace in this and the `full` classnames, just join the three on `" "` in your classname method.